notebooks/utils/utils.py

```diff
@@ -212,3 +215,29 @@ def print_key_metadata(ds, msg=None):
         print("ds['time']." + attr_name)
         print(getattr(ds["time"], attr_name))
         print(32 * "*")
+
+
+def generate_plot_catalog(root_dir, extension=".json"):
+    """
+    Generate a single dataframe from plot attributes saved in json files.
+    Parameters
+    ----------
+    root_dir : str, pathlib.Path
+          The root directory
+    extension : str, default `.json.`
+          file extension to look for.
+
+    Returns
+    -------
+    df : pd.DataFrame
+    """
+    root_dir = pathlib.Path(root_dir)
+    files = sorted(root_dir.rglob(f"**/*{extension}"))
+    data = []
+    if files:
+        for file in files:
+            data.append(json.load(file.open()))
+        return pd.DataFrame(data)
+    else:
+        print(f"Found 0 files with extension={extension} in {root_dir}.")
+        return pd.DataFrame()
```

Comment:
@mnlevy1981, so far this function returns the following

```python
In [1]: from utils import generate_plot_catalog

In [2]: root_dir = "/glade/work/mlevy/codes/hi-res_analysis/notebooks/images"

In [5]: df = generate_plot_catalog(root_dir)

In [6]: df.head()
Out[6]: 
                                            filepath  apply_log10        date          varname     plot_type
0  images/g.e22.G1850ECO_JRA_HR.TL319_t13.003/CaC...        False  0001-01-16  CaCO3_FLUX_100m  summary_plot
1  images/g.e22.G1850ECO_JRA_HR.TL319_t13.003/CaC...         True  0001-01-16  CaCO3_FLUX_100m  summary_plot
2  images/g.e22.G1850ECO_JRA_HR.TL319_t13.003/CaC...        False  0001-02-15  CaCO3_FLUX_100m  summary_plot
3  images/g.e22.G1850ECO_JRA_HR.TL319_t13.003/CaC...         True  0001-02-15  CaCO3_FLUX_100m  summary_plot
4  images/g.e22.G1850ECO_JRA_HR.TL319_t13.003/CaC...        False  0001-03-16  CaCO3_FLUX_100m  summary_plot
```

I haven't decided what to do with the filepaths yet